src/shared/calendar/calendar-provider.tsx

```diff
@@ -0,0 +1,52 @@
+import CalendarCells from './calendar-cells'
+import CalendarHeader from './calendar-header'
+import { CalendarProvider } from './hooks/useCalendarContext'
+import useCalendarDates from './hooks/useCalendarDate'
+import useCalendarNavigation from './hooks/useCalendarNavigation'
+import useSelectedDate from './hooks/useSelectedDate'
+import WeekdayHeader from './weekday-header'
+
+/**
+ * @description 캘린더 UI 컴포넌트
+ * CalendarProvider로 context를 하위 컴포넌트에 전달합니다.
+ *
+ * <하위컴포넌트>
+ * Calendar.Header - YYYY.MM 표시 헤더,
+ * Calendar.Weekdays - 일~토 요일 표시 헤더,
+ * Calendar.Cells - 이번 달 날짜
+ *
+ *
+ * @example
+ *    <Calendar>
+ *     <Calendar.Header />
+ *     <Calendar.Weekdays />
+ *     <Calendar.Cells />
+ *   </Calendar>
+ *
+ */
+
+export default function Calendar({ children }: { children: React.ReactNode }) {
```

Comment:
```suggestion
export default function Calendar({ children }: { children: ReactNode }) {
```

React 객체에서 마침표 표기법으로 접근할지, ReactNode만 불러와서 사용할건지는 정해져야할 것 같은데요. 유진님을 제외한 3명 모두 특정 타입, 메서드 들을 불러오도록 하고있어서 ReactNode만 불러오도록 변경해주셔야합니다!

> [!warning]
> 근데 React가 import된 적이없는데 이게 동작하나요 ?